### ISSUE TITLE
Unable to run `make deepcopy`

### ISSUE THREAD

**Issue by rlguarino** [GoogleCloudPlatform/metacontroller#79]
I'm unable to run `make` because the deepcopy target is failing.

```console
rssguar@cloudshell:~/gopath/src/k8s.io/metacontroller$ dep ensure --vendor-only
rssguar@cloudshell:~/gopath/src/k8s.io/metacontroller$ git s
rssguar@cloudshell:~/gopath/src/k8s.io/metacontroller$ make
+ Generating deepcopy funcs for metacontroller/v1alpha1
ERROR: logging before flag.Parse: F0822 17:04:10.405059     337 deepcopy.go:126] Failed loading boilerplate: open /home/rssguar/gopath/src/k8s.io/kubernetes/hack/boilerplate/boilerplate.go.txt: no such file or directory
goroutine 1 [running]:
k8s.io/metacontroller/vendor/github.com/golang/glog.stacks(0xc42000e000, 0xc42963dd40, 0xba, 0x10a)
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/github.com/golang/glog/glog.go:769 +0xcf
k8s.io/metacontroller/vendor/github.com/golang/glog.(*loggingT).output(0x8d5fa0, 0xc400000003, 0xc429726a50, 0x8b5122, 0xb, 0x7e, 0x0)
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/github.com/golang/glog/glog.go:720 +0x32d
k8s.io/metacontroller/vendor/github.com/golang/glog.(*loggingT).printf(0x8d5fa0, 0x3, 0x712bad, 0x1e, 0xc4270cfcf8, 0x1, 0x1)
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/github.com/golang/glog/glog.go:655 +0x14b
k8s.io/metacontroller/vendor/github.com/golang/glog.Fatalf(0x712bad, 0x1e, 0xc4270cfcf8, 0x1, 0x1)
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/github.com/golang/glog/glog.go:1148 +0x67
k8s.io/metacontroller/vendor/k8s.io/gengo/examples/deepcopy-gen/generators.Packages(0xc4296e8d80, 0xc4200a6960, 0x6e1f14, 0x6, 0xc4296e8d80)
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:126 +0xc5
k8s.io/metacontroller/vendor/k8s.io/gengo/args.(*GeneratorArgs).Execute(0xc4200a6960, 0xc420074db0, 0x6e1f14, 0x6, 0x71e060, 0x0, 0x0)
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/k8s.io/gengo/args/args.go:193 +0x1b8
main.main()
        /home/rssguar/gopath/src/k8s.io/metacontroller/vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go:69 +0x1c4
Makefile:28: recipe for target 'deepcopy' failed
make: *** [deepcopy] Error 255
rssguar@cloudshell:~/gopath/src/k8s.io/metacontroller$ git show
commit 2a992e2bb526749a6f10cad9b570fbe0510599f5
Merge: 98b9014 8922193
Author: Anthony Yeh <enisoc@google.com>
Date:   Mon Jul 23 15:05:26 2018 -0700

    Merge pull request #66 from crimsonfaith91/one-mc

    only allow one Metacontroller instance to be active at a time

rssguar@cloudshell:~/gopath/src/k8s.io/metacontroller$ git status
```

Is this just me doing something strange? I was going to take a look at #47 but I can't confirm I'm in a working state to being with. I noticed we never actually run `make` in CI so it might not work anymore.

**Comment by rlguarino** [comment]
I think in theory I should be able to make this work if I checkout the right version of Kubernetes on my gopath, any idea why it's not using the version that's vendored?

**Comment by rlguarino** [comment]
From my googling and crappy tests it seems to resolve itself if I use newer version of all of the libraries, might now be worth fixing depending on how fast #47 happens.

**Comment by enisoc** [comment]
I ran into this before too. It seems to be a known limitation of the older codegen that you have to have the core k8s repo in your gopath too so that it can read the boilerplate. As you mentioned, they fixed this in later codegen versions so this problem should be fixed as part of #80 as long as your prune.project suggestion works to retain the boilerplate file.
